src/Marten/Linq/MartenQueryExecutor.cs

```diff
@@ -49,35 +49,16 @@ private IResolver<T> resolver<T>()
             return _schema.StorageFor(typeof (T)).As<IResolver<T>>();
         }
             
-            
         T IQueryExecutor.ExecuteScalar<T>(QueryModel queryModel)
         {
-            var mapping = _schema.MappingFor(queryModel.SelectClause.Selector.Type);
-            var documentQuery = new DocumentQuery(mapping, queryModel, _expressionParser);
-
-            _schema.EnsureStorageExists(mapping.DocumentType);
-
-            if (queryModel.ResultOperators.OfType<AnyResultOperator>().Any())
-            {
-                var anyCommand = new NpgsqlCommand();
-                documentQuery.ConfigureForAny(anyCommand);
-
-                return _runner.Execute(anyCommand, c => (T)c.ExecuteScalar());
-            }
-
-            if (queryModel.ResultOperators.OfType<CountResultOperator>().Any())
-            {
-                var countCommand = new NpgsqlCommand();
-                documentQuery.ConfigureForCount(countCommand);
-
-                return _runner.Execute(countCommand, c =>
-                {
-                    var returnValue = c.ExecuteScalar();
-                    return Convert.ToInt32(returnValue).As<T>();
-                });
-            }
-
-            throw new NotSupportedException();
```

Comment:
Should I move this up into a field by replacing the whole generic results with an object?